drivers/led/led_npm1300.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_npm1300_led
+
+#include <errno.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/drivers/mfd/npm1300.h>
+#include <zephyr/dt-bindings/led/npm1300.h>
+
+/* nPM1300 LED base address */
+#define NPM_LED_BASE 0x0AU
+
+/* nPM1300 LED register offsets */
+#define NPM_LED_OFFSET_MODE 0x00U
+#define NPM_LED_OFFSET_SET  0x03U
+#define NPM_LED_OFFSET_CLR  0x04U
+
+/* nPM1300 Channel counts */
+#define NPM1300_LED_PINS 3U
+
+struct led_npm1300_config {
+	const struct device *mfd;
+	uint8_t mode[NPM1300_LED_PINS];
+};
+
+int led_npm1300_on(const struct device *dev, uint32_t led)
+{
+	const struct led_npm1300_config *config = dev->config;
+
+	if (led >= NPM1300_LED_PINS) {
+		return -EINVAL;
+	}
+
+	if (config->mode[led] != NPM1300_LED_HOST) {
+		return -EPERM;
+	}
+
+	return mfd_npm1300_reg_write(config->mfd, NPM_LED_BASE, NPM_LED_OFFSET_SET + (led * 2U),
+				     1U);
+}
+
+int led_npm1300_off(const struct device *dev, uint32_t led)
+{
+	const struct led_npm1300_config *config = dev->config;
+
+	if (led >= NPM1300_LED_PINS) {
+		return -EINVAL;
+	}
+
+	if (config->mode[led] != NPM1300_LED_HOST) {
+		return -EPERM;
+	}
+
+	return mfd_npm1300_reg_write(config->mfd, NPM_LED_BASE, NPM_LED_OFFSET_CLR + (led * 2U),
+				     1U);
+}
+
+static const struct led_driver_api led_npm1300_api = {
+	.on = led_npm1300_on,
+	.off = led_npm1300_off,
+};
+
+static int led_npm1300_init(const struct device *dev)
+{
+	const struct led_npm1300_config *config = dev->config;
+	int led;
+	int ret;
+
+	if (!device_is_ready(config->mfd)) {
+		return -ENODEV;
+	}
+
+	for (led = 0; led < NPM1300_LED_PINS; led++) {
+		if (config->mode[led] <= NPM1300_LED_HOST) {
+			ret = mfd_npm1300_reg_write(config->mfd, NPM_LED_BASE,
```

Comment:
ret can be scoped at this level

Reply:
In addition `ret` is not used at all in this function. Please return an error if the initialization fails.